src/platform/window.cpp

```diff
@@ -116,6 +141,36 @@ void Window::setKeyCallback(const KeyCallback &keyCallback) {
 	_keyCallback = keyCallback;
 }
 
+void Window::setMouseCallback(const MouseCallback &mouseCallback) {
+	_mouseCallback = mouseCallback;
+}
+
+void Window::setMousePositionCallback(const Axis2DCallback &mousePositionCallback) {
+	_mousePositionCallback = mousePositionCallback;
+}
+
+void Window::setMouseScrollCallback(const Axis2DCallback &mouseScrollCallback) {
+	_mouseScrollCallback = mouseScrollCallback;
+}
+
+void Window::lockMouse() {
+	GLFWwindow *window = this->getWindowHandle();
+	glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_DISABLED);
+	if (glfwRawMouseMotionSupported())
+    	glfwSetInputMode(window, GLFW_RAW_MOUSE_MOTION, GLFW_TRUE);
```

Comment:
According to [GLFW docs](https://www.glfw.org/docs/latest/input_guide.html#raw_mouse_motion), raw mouse motion (when available) tries to obtain movement information without OS's scaling/acceleration, so that used mouse movements could be more precise. One of the example use cases there includes a 3D camera, and that's why I ended up using it here. Should I place a comment there to make those manipulations clearer when browsing the code?